libcaf_test/caf/test/runner.cpp

```diff
@@ -68,7 +67,15 @@ int runner::run(int argc, char** argv) {
     return EXIT_SUCCESS;
   }
   default_reporter->start();
+  auto enabled = [](const std::regex& selected,
+                    const std::string_view& search_string) {
+    return std::regex_search(search_string.begin(), search_string.end(),
+                             selected);
+  };
+  auto suite_regex = parse_regex_opt(get_or(cfg_, "suites", ".*"));
   for (auto& [suite_name, suite] : suites_) {
+    if (suite_regex.has_value() && !enabled(suite_regex.value(), suite_name))
+      continue;
```

Comment:
That's a bit on the verbose side. We usually just use the automatic conversion to `bool` and `operator*`. As stated above, the function should return on an invalid configuration. So here, we know that the regex is valid:

```suggestion
    if (!enabled(*suite_regex, suite_name))
      continue;
```